dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/tools/DependentTestFinder.java

```diff
@@ -197,15 +198,55 @@ public static void runDTF(String dependentTestName, RESULT dependentTestResult,
 
         checkOrder(isOriginalOrder, dependentTestName, beforeTests, afterTests);
     }
+    //overloaded
+    public void runDTF(String dependentTestName, RESULT dependentTestResult, List<String> currentOrderTests,
+            List<String> originalOrderTests, List<String> filesToDelete, List<String> allDTList, String tmptestpath) {
+        DEPENDENT_TEST_RESULT = dependentTestResult;
+        CURRENT_ORDER_TESTS = currentOrderTests;
+        ORIGINAL_ORDER_TESTS = originalOrderTests;
+        FILES_TO_DELETE = filesToDelete;
+
+        if (allDTList == null) {
+            ALL_DT_LIST = new ArrayList<String>();
+        } else {
+            ALL_DT_LIST = allDTList;
+        }
+        boolean isOriginalOrder = isTestResultDifferent(dependentTestName, Arrays.asList(dependentTestName), tmptestpath);
+        List<String> fullTests;
+        if (isOriginalOrder) {
+            fullTests = ORIGINAL_ORDER_TESTS;
+        } else {
+            fullTests = CURRENT_ORDER_TESTS;
+        }
+
+        List<String> tests = new LinkedList<String>(fullTests);
+        if (tests == null || !tests.contains(dependentTestName)) {
+            throw new RuntimeException("Dependent test missing from test order.");
+        }
 
-    public static List<String> getAllDTs() {
+        List<String> beforeTests = new ArrayList<String>();
+        List<String> afterTests = new ArrayList<String>();
+        int testNameIndex = ALL_DT_LIST.indexOf(Constants.TEST_LINE + dependentTestName);
+        if (testNameIndex != -1) {
+            isOriginalOrder = determineSubLists(fullTests, testNameIndex, dependentTestName, isOriginalOrder, tests,
+                    beforeTests, afterTests, tmptestpath);
+        } else {
+            tests = fullTests.subList(0, fullTests.indexOf(dependentTestName));
+            dependentTestSolver(tests, dependentTestName, isOriginalOrder, beforeTests, afterTests,
+                    new ArrayList<String>(), new ArrayList<String>(), false, tmptestpath);
+        }
+
+        checkOrder(isOriginalOrder, dependentTestName, beforeTests, afterTests);        
+    }
+
+    public List<String> getAllDTs() {
         return ALL_DT_LIST;
     }
 
     // determines a sublist to look for the test(s) dependentTest depends on
     // returns true if the sublist is part of ORIGINAL_ORDER_TESTS,
     // false if the sublist is part of CURRENT_ORDER_TESTS
-    private static boolean determineSubLists(List<String> fullTests, int testNameIndex, String dependentTestName,
+    private boolean determineSubLists(List<String> fullTests, int testNameIndex, String dependentTestName,
```

Comment:
Please delete the body of this method and have this method call your new determineSubLists method with "" as the append argument.